flyteadmin/pkg/repositories/models/launch_plan.go

```diff
@@ -29,8 +36,9 @@ type LaunchPlan struct {
 	// GORM doesn't save the zero value for ints, so we use a pointer for the State field
 	State *int32 `gorm:"default:0"`
 	// Hash of the launch plan
-	Digest       []byte
-	ScheduleType LaunchPlanScheduleType
+	Digest              []byte
+	ScheduleType        LaunchPlanScheduleType
+	LaunchConditionType *LaunchConditionType
```

Comment:
just curious, should this be indexed?

Reply:
yeah i added a conditional index in the migration - https://github.com/flyteorg/flyte/pull/4903/files#diff-7189b1a0f8ed581ca15085037de7cd70b60305b90e42cc60f7eda13796577788R1221.  is that okay?

Reply:
maybe add here for consistency?

Reply:
let me try...

Reply:
k - i think that is it https://github.com/flyteorg/flyte/pull/4903/files#diff-34110b943c015fdb7456f2c8fa9759e25ff476bf32ed261b2945f7bb928f56dcR41.

but i kinda want to keep the explicit index in the migration if that's okay?

Reply:
actually, nm, it's cleaner to just use gorm, as much as i dislike it.  I added a type, dropped the column in my local sandbox postgres and re-ran the migration with the new annotation.  Gorm ran these two queries.
```
ALTER TABLE "launch_plans" ADD "launch_condition_type" varchar(32)
```

and

```
 CREATE INDEX IF NOT EXISTS "idx_launch_plans_launch_conditions" ON "launch_plans" ("launch_condition_type") WHERE launch_condition_type is not null
```